### PR TITLE
Added back metrics support for extensions

### PR DESCRIPTION
**Pull request by chiayi**
Add back metrics support within extensions

**What type of PR is this?**
> Uncomment only one ` /kind <>` line, press enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind breaking
> /kind bug
> /kind cleanup
> /kind documentation
> /kind feature

/kind hotfix

**What this PR does / Why we need it**:
This adds back the metrics support within `Extensions` that was removed during the trimming process.

**Which issue(s) this PR fixes**:
Work on https://github.com/googleforgames/agones/issues/2797

**Special notes for your reviewer**:

### CODE REVIEW

File: cmd/extensions/main.go [hunk 1]
```diff
@@ -139,8 +144,44 @@ func main() {
...
+		if err != nil {
+			logger.WithError(err).Fatal("Could not register stackdriver exporter")
+		}
+		// It is imperative to invoke flush before your main function exits
+		defer sd.Flush()
+	}
+
+	// Prometheus metrics
+	if ctlConf.PrometheusMetrics {
+		logger.Info("registering prometheus metrics...")
```
Comment (zmerlynn) [inline cmd/extensions/main.go:162]: let's take the logging back out and make this symmetric with controller

File: cmd/extensions/main.go [hunk 2]
```diff
@@ -139,8 +144,44 @@ func main() {
...
 
-	health := healthcheck.NewHandler()
+	// If we are using Prometheus only exporter we can make reporting more often,
+	// every 1 seconds, if we are using Stackdriver we would use 60 seconds reporting period,
+	// which is a requirements of Stackdriver, otherwise most of time series would be invalid for Stackdriver
+	metrics.SetReportingPeriod(ctlConf.PrometheusMetrics, ctlConf.Stackdriver)
+
+	// Add metrics controller only if we configure one of metrics exporters
+	if ctlConf.PrometheusMetrics || ctlConf.Stackdriver {
+		rs = append(rs, metrics.NewController(kubeClient, agonesClient, kubeInformerFactory, agonesInformerFactory))
```
Comment (zmerlynn) [inline cmd/extensions/main.go:183]: There's a difference between exporting metrics about this process, which is what line 160-174 are doing, and exporting metrics about the fleet / etc. The "metrics controller" is doing the latter, and I don't think we need two instances of it running, so I'm _pretty sure_ you can safely kill 182-184 here, which means you also don't need `rs`.

**Review by zmerlynn — CHANGES_REQUESTED** [review]

File: cmd/extensions/main.go (hunk 1 above)
Comment (chiayi) [reply cmd/extensions/main.go]: done.

File: cmd/extensions/main.go (hunk 2 above)
Comment (chiayi) [reply cmd/extensions/main.go]: okay! Done!

**Review by zmerlynn — APPROVED** [review]

**Pull request merged**